integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/AbstractCdiTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2000-2018 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.quarkus.it;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+import org.junit.Assert;
+import org.openqa.selenium.By;
+
+import com.vaadin.flow.test.AbstractChromeTest;
+
+abstract public class AbstractCdiTest extends AbstractChromeTest {
+
+    protected void click(String elementId) {
+        findElement(By.id(elementId)).click();
+    }
+
+    protected void follow(String linkText) {
+        findElement(By.linkText(linkText)).click();
+    }
+
+    protected String getText(String id) {
+        return findElement(By.id(id)).getText();
+    }
+
+    protected void assertCountEquals(int expectedCount, String counter)
+            throws IOException {
+        Assert.assertEquals(expectedCount, getCount(counter));
+    }
+
+    protected void assertTextEquals(String expectedText, String elementId) {
+        Assert.assertEquals(expectedText, getText(elementId));
+    }
+
+    protected void resetCounts() throws IOException {
+        slurp("?resetCounts");
+    }
+
+    protected int getCount(String id) throws IOException {
+        getCommandExecutor().waitForVaadin();
+        String line = slurp("?getCount=" + id);
+        return Integer.parseInt(line);
+    }
+
+    private String slurp(String uri) throws IOException {
```

Comment:
slurp? This naming is from clojure? also slurp should read the whole file, but in this instance we only read the first line in the response. Also it doesn't take a uri but queryParameters for the CounterFilter.

 Perhaps another name would be more appropriate to lessen confusion on what it does, like `doAction(String parameter)` or something.

Reply:
The method comes from https://github.com/vaadin/cdi/blob/master/vaadin-cdi-itest/src/test/java/com/vaadin/cdi/itest/AbstractCdiTest.java.
This is just a copy .
I don't want to rename the method even though I don't like the name. It's better to keep the consistency between original code and copy.